flink-runtime/src/main/java/org/apache/flink/runtime/entrypoint/component/DefaultDispatcherResourceManagerComponentFactory.java

```diff
@@ -71,24 +71,23 @@
 /**
  * Abstract class which implements the creation of the {@link DispatcherResourceManagerComponent} components.
  *
- * @param <T> type of the {@link DispatcherRunner}
  * @param <U> type of the {@link RestfulGateway} given to the {@link WebMonitorEndpoint}
  */
-public abstract class AbstractDispatcherResourceManagerComponentFactory<T extends DispatcherRunner, U extends RestfulGateway> implements DispatcherResourceManagerComponentFactory {
+public class DefaultDispatcherResourceManagerComponentFactory<U extends RestfulGateway> implements DispatcherResourceManagerComponentFactory {
```

Comment:
I don't think one should instance an {{DefaultDispatcherResourceManagerComponentFactory}}. How about keep it `abstract`? Both names are OK for me.

Reply:
Why not? The `DefaultDispatcherResourceManagerComponentFactory` has no abstract methods anymore. Moreover, the `JobDispatcherResourceManagerComponentFactory` and `SessionDispatcherResourceManagerComponentFactory` are only used to set the factory up in a certain way. I think they could actually be removed and replaced by some static helper methods. There is no need to have different subclasses for this.

Reply:
Yes I agree we can have one dispatcher rm component and refactor the difference into helpers. If it's on the way I'm ok with this temporary state. Otherwise if one should always instance a job/session subclass it's better to mark as abstract.